packages/op-coreutils/src/op_coreutils/bigquery/write.py

```diff
@@ -212,6 +201,40 @@ def overwrite_partition_static(
     )
 
 
+@dataclass
+class DatePart:
+    date_df: pl.DataFrame
+    date_suffix: str
+
+
+def breakout_partitioned_df(df: pl.DataFrame) -> Generator[DatePart, None, None]:
+    """Checks that the dataframe is suitable for writing to a partitioned table.
+
+    Yields each date part along with the date part suffix for each date present in
+    the data.
+    """
+
+    # Ensure "dt" is a DateTime
+    if df["dt"].dtype == pl.String:
+        df = df.with_columns(dt=pl.col("dt").str.strptime(pl.Datetime, "%Y-%m-%d"))
+
+    partitions = df["dt"].unique().sort().to_list()
+
+    log.info(
+        f"Found {len(partitions)} partitions in dataframe [{partitions[0]} ... {partitions[-1]}]"
+    )
+
+    if len(partitions) > 10:
```

Comment:
maybe in the future we can make this a parameter instead of limiting it to 10 partitions. For now I don't see cases where we might need to update that many partitions at once. It would defeat the purpose of incremental processing.

Reply:
Agreed, adding a todo: comment here so that we keep track of this.